lib/core/src/Cardano/Wallet/Primitive/Types/Coin/Gen.hs

```diff
@@ -19,7 +20,11 @@ import Test.QuickCheck
 --------------------------------------------------------------------------------
 
 genCoin :: Gen Coin
-genCoin = sized $ \n -> Coin . fromIntegral <$> choose (0, n)
+genCoin = frequency
+    [ (1, pure (Coin 0))
+    , (5, Coin <$> choose (1_000_000, 1_000_000_000_000))
+    ]
```

Comment:
Hi @Anviking 

We recently pushed a series of PRs to convert many generators (such as this one) to use the QC size parameter.

I think using the size parameter is a good idea, since:
- for many tests of pure algorithms (coin selection, migration), we just need small-ish values.
- QC will adjust this parameter automatically over several runs to provide a reasonable distribution.

For tests that specifically need very large values, or special ranges, we provide dedicated generators.

If we're going to change direction (and change generators such as this one so that they no longer use the size parameter), then would you be okay with us having a discussion about it? We could discuss this in the development meeting.

Reply:
I ended up not having to modify this Arbitrary instance.